### PR TITLE
Fix app upgrade for releave/0.9.18

### LINKED ISSUE #4077: `JavascriptException: Migration is required due to the following errors` is shown after upgrading 0.9.17 to 0.9.18
### Description
The exception is shown only for Android. The app crashes on IOS after upgrading and the exception is not shown

*Type*: Bug

#### Expected behavior
An alert about excluded upgrade option from the release is shown, no crashes, user can create new account or recover the old one
screenshot_2018-05-01-18-21-30

#### Actual behavior
The error is shown, the app crashes if user taps on 'EXIT' bbutton
screenshot_2018-05-01-17-52-38

### Reproduction
0.9.17 is installed, an account is created
upgrade 0.9.17 to [0.9.18 80a0bc](http://artifacts.status.im:8081/artifactory/nightlies-local/im.status.ethereum-80a0bc.apk) version 

### Additional Information
* Status version: upgrade 0.9.17 to [0.9.18 80a0bc](http://artifacts.status.im:8081/artifactory/nightlies-local/im.status.ethereum-80a0bc.apk)
* Operating System: Android and iOS

### PR DESCRIPTION
**Pull request by janherich**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

fixes #4077 

### Steps to test:
Try to install the build over `0.9.17` release build

status: ready

### CODE REVIEW

File: src/status_im/data_store/realm/core.cljs [hunk 1]
```diff
@@ -49,11 +49,9 @@
   ;; TODO: remove for release 0.9.18
   ;; delete the realm file if its schema version is higher
   ;; than existing schema version (this means the previous
-  ;; install has incompatible database schemas)
-  (if (> (realm-version file-name)
-         (apply max (map :schemaVersion base/schemas)))
-    (reset-realm file-name schemas)
-    (migrate-realm file-name schemas)))
+  ;; install has incompatible database schemas) 
+  (reset-realm file-name schemas)
```
Comment (yenda) [inline src/status_im/data_store/realm/core.cljs]: have you tested this ? To me it seems like it will just delete the db everytime you start the app ?

Comment (janherich) [reply src/status_im/data_store/realm/core.cljs]: You are correct

**Comment by janherich** [comment]
@yenda I fixed the logic:
I added dummy `v2` schema versions to `release/0.9.18`, and added condition that realm purge will be performed whenever installed schema version is lower then highest schema version in codebase (which will be the case for `0.9.17` -> `0.9.18` install (`v1` -> `v2` change)).
From now on we should probably restore the good practice of introducing realm changes in migrations and start with **v3** (so skip **v2** in develop), as **v2** will be the schema version of the newest release (`0.9.18`) app.

**Comment by yenda** [comment]
@janherich I think it would be cleaner to have a file written the first time when you reset.

**Comment by janherich** [comment]
@yenda you mean some special "lock" file ?

**Comment by yenda** [comment]
yes for instance v18.lock, reset if not present

**Comment by janherich** [comment]
@yenda I don't particularly like the idea of some another persistance source other then realm, with it's own write/read API, even if it's just simple flat file.
The solution with Version jump may sound hacky (it is), but it's the simplest thing to do and once V3 migration lands in develop, we can completely forget about it.

**Review by goranjovic — APPROVED** [review]

**Comment by antdanchenko** [comment]
An error is shown after upgrading 0.9.17 with [PR build](https://i.diawi.com/kVRZDR) on Android

screenshot_2018-05-02-16-59-09

An error is shown after upgrading 0.9.17 with [PR build](https://i.diawi.com/MyUko2) on IOS

image_uploaded_from_ios

After pressing 'OK' a white screen is shown for both platforms

**Steps to reproduce:**
Install 0.9.17 
Create an account on 0.9.17 version
Upgrade 0.9.17 with the PR build

**Review by yenda — APPROVED** [review]

**Comment by antdanchenko** [comment]
### Actual Result:
App hangs on `Signing you in..` after updating 0.9.17 to 0.9.18 '3451e9' (android only)

image

### Steps to Reproduce:

- Install 0.9.17 from PlayMarket and create an account 
- Upgrade 0.9.17 to 0.9.18 '3451e9
- Open the app and enter valid password and tap on sign in

@janherich It won't be easy to introduce an upgrade in 0.9.18 release, also with https://github.com/status-im/ideas/tree/master/ideas/142-recovery-compatibility we will exclude it one more time in 0.9.19 or 0.9.20

It seems that the best way is to implement `Expected behavior` from https://github.com/status-im/status-react/issues/4077 (An alert about excluded upgrade option from the release is shown, no crashes, user can create new account or recover the old one)

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/status-im/ideas">status-im/ideas</a></strong></div><div>ideas - Swarm Home. New, completed and in-progress features for Status</div></blockquote>

**Pull request merged**